pkg/engine/engine.go

```diff
@@ -73,94 +79,147 @@ func (e *QueryEngine) Query(params logql.Params) logql.Query {
 //  2. Create a physical plan from the logical plan using information from the catalog.
 //  3. Evaluate the physical plan with the executor.
 func (e *QueryEngine) Execute(ctx context.Context, params logql.Params) (logqlmodel.Result, error) {
-	start := time.Now()
+	ctx, span := tracer.Start(ctx, "QueryEngine.Execute", trace.WithAttributes(
+		attribute.String("query", params.QueryString()),
```

Comment:
would be nice to also include start, end, range_type and step similar to existing engine
https://github.com/grafana/loki/blob/98f7d8198672a88c3e8d571e90fcf580a0f7db4c/pkg/logql/engine.go#L273